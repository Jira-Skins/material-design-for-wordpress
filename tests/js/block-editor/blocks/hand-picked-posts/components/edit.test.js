/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import Edit from '../../../../../../assets/src/block-editor/blocks/hand-picked-posts/edit';

// Mock PostsControl component as not relevant in this test and failing to pass tests
// due to the SearchListControl WooCommerce component.
jest.mock(
	'../../../../../../assets/src/block-editor/components/posts-control/index.js',
	() => {
		const PostsControl = () => <div />;
		return PostsControl;
	}
);

/**
 * Shallow render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return shallow( <Edit { ...props } /> );
};

// TODO: This test needs to be revised as this is probably not the correct way to do it but serves as a base.
// The problem with testing this component is due to hooks in child components.
describe( 'Edit', () => {
	it( 'matches snapshot', () => {
		const wrapper = setup( {
			attributes: {
				style: 'masonry',
				contentLayout: 'text-above-media',
				columns: 3,
				postsToShow: 12,
				outlined: false,
				displayPostDate: true,
				displayPostContent: true,
				postContentLength: 20,
				displayFeaturedImage: true,
				displayCommentsCount: true,
				displayPostAuthor: false,
				category: undefined,
				posts: [],
				editMode: true,
			},
		} );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
