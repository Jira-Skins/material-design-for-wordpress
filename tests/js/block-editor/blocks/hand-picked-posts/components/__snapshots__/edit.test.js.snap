// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit matches snapshot 1`] = `
<Edit
  attributes={
    Object {
      "category": undefined,
      "columns": 3,
      "contentLayout": "text-above-media",
      "displayCommentsCount": true,
      "displayFeaturedImage": true,
      "displayPostAuthor": false,
      "displayPostContent": true,
      "displayPostDate": true,
      "editMode": true,
      "outlined": false,
      "postContentLength": 20,
      "posts": Array [],
      "postsToShow": 12,
      "style": "masonry",
    }
  }
  debouncedSpeak={[Function]}
  speak={[Function]}
/>
`;
